tests/network/primary_pod_network.go

```diff
@@ -85,60 +75,105 @@ var _ = SIGDescribe("Primary Pod Network", func() {
 			var vmi *v1.VirtualMachineInstance
 
 			BeforeEach(func() {
-				vmi = setupVMI(virtClient, vmiWithMasqueradeBinding())
+				vmi = libvmi.SetupVMI(vmiWithMasqueradeBinding())
 			})
 
 			AfterEach(func() {
-				cleanupVMI(virtClient, vmi)
+				libvmi.CleanupVMI(vmi)
 			})
 
 			It("should report PodIP as its own on interface status", func() { AssertReportedIP(vmi) })
 		})
 	})
-})
 
-func setupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
-	By("Creating the VMI")
-	var err error
-	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	Describe("Connectivity", func() {
+		const servicePort = 1500
 
-	By("Waiting until the VMI gets ready")
-	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInAlpineExpecter)
+		AssertConnectivity := func(outboundVMI, inboundVMI *v1.VirtualMachineInstance) {
+			inboundVMIAddress := inboundVMI.Status.Interfaces[0].IP
 
-	return vmi
-}
+			By("Checking connectivity via ICMP")
+			Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress)).To(Succeed())
 
-func cleanupVMI(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) {
-	if vmi != nil {
-		By("Deleting the VMI")
-		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
-
-		By("Waiting for the VMI to be gone")
-		Eventually(func() error {
-			_, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(vmi.GetName(), &metav1.GetOptions{})
-			return err
-		}, 2*time.Minute, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
-	}
-}
+			By("Checking connectivity via ICMP with MTU-sized frames")
+			mtu := getPodNetworkMTU(outboundVMI)
+			ipHeaderSize := 28 // IPv4 specific
+			Expect(tests.PingFromVMConsole(outboundVMI, inboundVMIAddress, "-c1", "-w5", "-s", strconv.Itoa(mtu-ipHeaderSize))).To(Succeed())
+
+			By("Checking connectivity via HTTP")
+			Expect(tests.PingHTTPFromConsole(outboundVMI, inboundVMIAddress, servicePort)).To(Succeed())
+		}
+
+		Context("2 VMIs connected to the pod network using the bridge binding", func() {
+			vmis := libvmi.NewVMISet()
+
+			BeforeEach(func() {
+				vmis["outbound"] = vmiWithBridgeBinding()
+				vmis["inbound"] = vmiWithBridgeBinding()
+				vmis.Setup()
+				tests.StartHTTPServer(vmis["inbound"], servicePort, false)
+			})
+
+			AfterEach(func() {
+				vmis.Cleanup()
+			})
+
+			It("[test_id:1540]should be able to reach from one to another", func() { AssertConnectivity(vmis["outbound"], vmis["inbound"]) })
+		})
+
+		Context("2 VMIs connected to the pod network using the masquerade binding", func() {
```

Comment:
This ones feels like a pair of table teste
```golang
Entry("with masquerade binding should be able to reach from one to another", vmiWithMasqueradeBinding)
Entry("with bridge binding should be able to reach from one to another", vmiWithBridgeBinding)
```

Reply:
DesribeTable is terrible when it comes for e2e tests. It would not allow us to use BeforeEach and AfterEach. And we don't want to prepare resources inside the test, since it would cause issues with cleanup after a failure.